test/Microsoft.AspNetCore.Server.Kestrel.FunctionalTests/ResponseTests.cs

```diff
@@ -321,14 +320,15 @@ public Task ResponseStatusCodeSetBeforeHttpContextDisposedRequestMalformedReadIg
                 }
                 else
                 {
-                    using (var socket = new Socket(AddressFamily.InterNetwork, SocketType.Stream, ProtocolType.Tcp))
+                    using (var connection = new TestConnection(host.GetPort()))
                     {
-                        socket.Connect(new IPEndPoint(IPAddress.Loopback, host.GetPort()));
-                        socket.Send(Encoding.ASCII.GetBytes(
-                            "POST / HTTP/1.1\r\n" +
-                            "Transfer-Encoding: chunked\r\n" +
-                            "\r\n" +
-                            "wrong"));
+                        await connection.Send(
+                            "POST / HTTP/1.1",
+                            "Transfer-Encoding: chunked",
+                            "",
+                            "wrong");
```

Comment:
Given the [failures on AppVeyor](https://ci.appveyor.com/project/aspnetci/kestrelhttpserver/build/1.0.3846). I would try changing "wrong" to "wr" to ensure connection.Send doesn't fail due to an aborted connection. This would be similar to the change I made in #1190.

I'm not sure what to do about Receive failing. Maybe call Receive before Send?
    Microsoft.AspNetCore.Server.Kestrel.FunctionalTests.ResponseTests.ResponseStatusCodeSetBeforeHttpContextDisposedRequestMalformedRead [FAIL]
      System.IO.IOException : Unable to write data to the transport connection: An established connection was aborted by the software in your host machine.
      ---- System.Net.Sockets.SocketException : An established connection was aborted by the software in your host machine
      Stack Trace:
           at System.Net.Sockets.NetworkStream.BeginWrite(Byte[] buffer, Int32 offset, Int32 size, AsyncCallback callback, Object state)
           at System.Net.Sockets.NetworkStream.<>c.<WriteAsync>b__59_0(Byte[] bufferArg, Int32 offsetArg, Int32 sizeArg, AsyncCallback callback, Object state)
           at System.Threading.Tasks.TaskFactory`1.FromAsyncImpl[TArg1,TArg2,TArg3](Func`6 beginMethod, Func`2 endFunction, Action`1 endAction, TArg1 arg1, TArg2 arg2, TArg3 arg3, Object state, TaskCreationOptions creationOptions)
           at System.Threading.Tasks.TaskFactory.FromAsync[TArg1,TArg2,TArg3](Func`6 beginMethod, Action`1 endMethod, TArg1 arg1, TArg2 arg2, TArg3 arg3, Object state)
           at System.Net.Sockets.NetworkStream.WriteAsync(Byte[] buffer, Int32 offset, Int32 size, CancellationToken cancellationToken)
           at System.IO.StreamWriter.<FlushAsyncInternal>d__67.MoveNext()
        --- End of stack trace from previous location where exception was thrown ---
           at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
           at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
           at System.Runtime.CompilerServices.TaskAwaiter.GetResult()
        C:\projects\kestrelhttpserver\test\shared\TestConnection.cs(61,0): at Microsoft.AspNetCore.Testing.TestConnection.<Send>d__7.MoveNext()
        --- End of stack trace from previous location where exception was thrown ---
           at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
           at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
           at System.Runtime.CompilerServices.TaskAwaiter.GetResult()
        C:\projects\kestrelhttpserver\test\Microsoft.AspNetCore.Server.Kestrel.FunctionalTests\ResponseTests.cs(325,0): at Microsoft.AspNetCore.Server.Kestrel.FunctionalTests.ResponseTests.<ResponseStatusCodeSetBeforeHttpContextDispose>d__10.MoveNext()
        --- End of stack trace from previous location where exception was thrown ---
           at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
           at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
        --- End of stack trace from previous location where exception was thrown ---
           at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
           at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
        --- End of stack trace from previous location where exception was thrown ---
           at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
           at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
        ----- Inner Stack Trace -----
           at System.Net.Sockets.Socket.BeginSend(Byte[] buffer, Int32 offset, Int32 size, SocketFlags socketFlags, AsyncCallback callback, Object state)
           at System.Net.Sockets.NetworkStream.BeginWrite(Byte[] buffer, Int32 offset, Int32 size, AsyncCallback callback, Object state)